File: python/rna-transcription/rna_transcription.py
transcription = {
   "G": "C",
   "C": "G",
   "T": "A",
   "A": "U"
}


def to_rna(dna_strand):

    res = []
    for nucleotide in dna_strand:
        res.append(transcription[nucleotide])

    return "".join(res)

File: clojure/rna-transcription/src/rna_transcription.clj
(ns rna-transcription)

(defn get-complement [nucleotide]
  (case nucleotide
    \G \C
    \C \G
    \T \A
    \A \U
    (throw (AssertionError. "invalid"))
   )
  )

(defn to-rna [dna]
  (if
    (empty? dna) ""
    (apply str (get-complement (first dna)) (to-rna (rest dna)))
    )
)
